### DISCUSSION TITLE
How can I change 'post_logout_redirect_uri' to another?

### DISCUSSION THREAD

**Discussion by snowman-code** [OpenIDC/mod_auth_openidc#661] [Q&A]
Thank you, mod_auth_openidc works good.

Here's the question.

When IdP servier is not allow 'post_logout_redirect_uri' keyword because server redirect_uri is set 'service'. 
(CAS OAuth Server default value is 'service').

And server admin can't change 'service' to 'post_logout_redirect_uri'  because so many other clients already use 'service' for redirect_uri.

Is any other options without changing source code?

With many thanks,
Micheal

**Comment by zandbelt** [comment]
you can provide your own URL via the `logout` parameter when triggering logout, see: https://github.com/zmartzone/mod_auth_openidc/wiki#9-how-do-i-logout-users

**Reply by snowman-code** [reply]
More detail question.

such as, 
/redirect_uri?logout=http://domain/
is translated to 
http://~~~/logout?post_logout_redirect_uri=http://domain/
by mod_auth_oidc

but I need this
/logout?**service**=http://domain/

So, I tried this
/redirect_uri?logout=http://domain/&**service**=http://domain/
and
/redirect_uri?logout&**service**=http://domain/
but, both of them not worked.

So, if there is no way to change param,
then I have to customize your code like this,

"%s%spost_logout_redirect_uri=%s", logout_request,
to
"%s%sservice=%s", logout_request,

Am I right?
How about that?

**Reply by zandbelt** [reply]
you can redirect to a page that you have customised to redirect in its turn to the URL that you need to

**Reply by HoussemGargouri** [reply]
I have maybe the same problem : 

The URL of callback :

`https://front..../oidc-callback/logout=....`

redirect me to : 

`https://auth..../realms/myrealm/protocol/openid-connect/logout?post_logout_redirect_uri=....`

which depends the `id_token_hint`. There are no way to force the redirect me to ?

`https://auth..../realms/myrealm/protocol/openid-connect/logout?redirect_uri=....`

Thx !

**Reply by HoussemGargouri** [reply]
@snowman-code Did you resolve this pls ?
